connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -141,17 +143,38 @@ public void initialize(TaskConfig taskConfig) {
     public void stop() {
         // Offset commit is handled upon exit in work thread
         super.stop();
+        tryStop();
```

Comment:
This may be called from a different thread, when the worker attempts to stop the task. OTOH `close()` is always called within the `WorkerTask.run()` via `doRun()` that calls `doClose()` in its finally block, which calls `close()`. That means it's possible for two different threads to call the `tryStop()` method.

Why does `stop()` even call `tryStop()`? The way it currently works is this:
1. In this method, `super.stop()` calls `triggerStop()` that sets the `stopping` flag and calls `notifyAll()`
1. any waiting thread calling `run` and ultimately `execute()` wakes up, and the `execute()` method's loop will check the `stopping` flag and return from the `execute()`, at which point the `doClose()` method is called in the finally block that ultimately calls this class' `close()` method.

IOW, this `stop()` method doesn't actually stop the worker task, but marks it as _needing_ to be stopped. The execute method discovers this in its loop and exits, and `close()` will always be called when the task stops its execution loop.

Removing the `tryStop()` method from `stop()` also means that the `stopped` field is no longer read/written by multiple threads and therefore can remain not volatile.